apps/api/src/app/events/usecases/trigger-event/trigger-event.usecase.ts

```diff
@@ -175,4 +201,52 @@ export class TriggerEvent {
       );
     }
   }
+
+  private async getProviderIdsForTemplate(
+    userId: string,
+    organizationId: string,
+    environmentId: string,
+    template: NotificationTemplateEntity
+  ): Promise<Map<ChannelTypeEnum, string>> {
+    const providers = new Map<ChannelTypeEnum, string>();
+
+    for (const step of template?.steps) {
+      const type = step.template?.type;
+      if (type) {
+        const channelType = STEP_TYPE_TO_CHANNEL_TYPE.get(type);
+
+        if (channelType) {
+          const provider = await this.getProviderId(userId, organizationId, environmentId, channelType);
+          if (provider) {
+            providers.set(channelType, provider);
+          } else {
+            providers.set(channelType, InAppProviderIdEnum.Novu);
```

Comment:
I am keeping this for retrocompatibility but can't grasp the need of setting a default provider if no provider for the channel. Is related to our free trial provider? @davidsoderberg

Reply:
I am not sure (i do think we need to add the novu integration by default as the in-app channel, that way we can use it more securely and not as a default)

Small note maybe we should add it only if it is in-app channelType. (i am pretty sure it could be otherwise but still)

Reply:
Should I add that check? Do not mind if we find it is safe to do. If not, I'd rather leave it for a different PR.

Reply:
Tend to agree it was a mall thought I had.
That's why I think we need to treat 'novu' in-app as a regular integration.